beacon_node/beacon_chain/src/execution_payload.rs

```diff
@@ -81,14 +81,10 @@ impl<T: BeaconChainTypes> PayloadNotifier<T> {
 
             match notify_execution_layer {
                 NotifyExecutionLayer::No if chain.config.optimistic_finalized_sync => {
-                    // Verify the block hash here in Lighthouse and immediately mark the block as
-                    // optimistically imported. This saves a lot of roundtrips to the EL.
-                    let execution_layer = chain
-                        .execution_layer
-                        .as_ref()
-                        .ok_or(ExecutionPayloadError::NoExecutionConnection)?;
-
-                    if let Err(e) = execution_layer.verify_payload_block_hash(block_message) {
+                    // Create a NewPayloadRequest (no clones required) and check optimistic sync verifications
+                    let new_payload_request: NewPayloadRequest<T::EthSpec> =
+                        block_message.try_into()?;
+                    if let Err(e) = new_payload_request.perform_optimistic_sync_verifications() {
```

Comment:
I notice we only do this verification when we're doing optimistic sync without the EL, yet the spec says we should always do it. I can't think of a good reason to _always_ do it, but do we know why the spec says that?

https://github.com/ethereum/consensus-specs/blob/v1.4.0-beta.2/specs/deneb/beacon-chain.md#modified-process_execution_payload

I noticed that the execution API specs say that the EL will always verify the versioned hashes, so it seems redundant for us to do it on the CL side as well

https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#engine_newpayloadv3